.github/workflows/mock-data.yml

```diff
@@ -0,0 +1,14 @@
+name: mock-data
+on:
+  push:
+    branches: [master, test-me-*]
+  pull_request:
+
+jobs:
+  mock-data:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - run: docker compose up --wait
+    - run: while ! curl http://localhost:3000; do sleep .5; done
```

Comment:
there's builtin health checking in docker-compose and docker -- but it's too clunky to use before docker 25 (unreleased)

Reply:
should we have a timeout?

Reply:
it's the line right below this one

Reply:
`--fail` is the easier way to do that